app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -54,6 +62,60 @@ def get_repository_ids(capsule, environment, content_view, repository)
           repository_ids
         end
 
+        def repos_needing_distributor_updates(capsule, environment, content_view)
+          repos = capsule.repos_available_to_capsule(capsule, environment, content_view)
+          update_distributors = []
+          repos.each do |repo|
+            generated_distributors = repo.generate_distributors
+            repo_details = capsule.pulp_repo_facts(repo.pulp_id)
+            next unless repo_details
+            capsule_distributors = repo_details["distributors"]
+            update_distributors << repo if distributors_changed?(capsule_distributors, generated_distributors)
+          end
+          update_distributors
+        end
+
+        def repos_needing_importer_updates(capsule, environment, content_view)
+          repos = capsule.repos_available_to_capsule(capsule, environment, content_view)
+          update_importer = []
+          repos.each do |repo|
+            repo_details = capsule.pulp_repo_facts(repo.pulp_id)
+            next unless repo_details
+            importer_config = repo_details["importers"][0]["config"]
+            importer = repo.generate_importer(capsule)
+            generated_importer_config = {"download_policy": importer.download_policy, "remove_missing": importer.remove_missing }.to_json
+            if generated_importer_config != importer_config.slice("download_policy", "remove_missing").to_json
+              update_importer << repo
+            end
+          end
```

Comment:
You could consider here, and above, extracting the logic into a function like `importer_changed?` that can either be called on a repository object or a repository object could be passed to. There is a lot of specific logic to how importers and distributors look and their attributes that might be better served captured in a different layer such as the Pulp service layer.